src/ipc-handlers.ts

```diff
@@ -778,22 +781,68 @@ export async function exportSiteToPush(
 		}
 
 		const stats = fs.statSync( archivePath );
-		const archiveContent = fs.readFileSync( archivePath );
-		return { archivePath, archiveContent, archiveSizeInBytes: stats.size };
+		return { archivePath, archiveSizeInBytes: stats.size };
 	} finally {
 		SYNC_ABORT_CONTROLLERS.delete( operationId );
 	}
 }
 
-export function removeTemporalFile( event: IpcMainInvokeEvent, path: string ) {
+export async function pushArchive(
+	event: IpcMainInvokeEvent,
+	remoteSiteId: number,
+	archivePath: string,
+	optionsToSync?: string[]
+): Promise< { success: boolean; error?: string } > {
+	const token = await getAuthenticationToken();
+
+	if ( ! token?.accessToken ) {
+		throw new Error( 'No token found' );
+	}
+
+	const wpcom = wpcomFactory( token.accessToken, wpcomXhrRequest );
+	const formData: [ string, unknown, Record< string, string >? ][] = [
+		[
+			'import',
+			fs.createReadStream( archivePath ),
+			{
+				filename: 'loca-env-site-1.tar.gz',
+				contentType: 'application/gzip',
+			},
+		],
+	];
+
+	if ( optionsToSync ) {
+		formData.push( [ 'options', optionsToSync.join( ',' ) ] );
+	}
+
+	try {
+		await wpcom.req.post( {
+			path: `/sites/${ remoteSiteId }/studio-app/sync/import`,
+			apiNamespace: 'wpcom/v2',
+			formData,
+		} );
+
+		return { success: true };
+	} catch ( error ) {
+		const parseResult = z.object( { error: z.string() } ).safeParse( error );
+
+		if ( parseResult.success ) {
+			return { success: false, error: parseResult.data.error };
+		}
+
+		return { success: false, error: 'Unknown error' };
```

Comment:
IPC functions that throw in the node process will also throw in the renderer process. The crux is that the error object thrown by the renderer has a different structure, so the most reliable way to pass data back to the renderer is to simply return.